### PR TITLE
[TACHYON-1428] Documented parts of the tachyon-servers module in preparation for the checkstyle rule

### PR DESCRIPTION
**Pull request by HorizonNet**
https://tachyon.atlassian.net/browse/TACHYON-1428

### CODE REVIEW

File: servers/src/main/java/tachyon/worker/block/BlockMetadataManager.java [hunk 1]
```diff
@@ -166,7 +166,7 @@ public void cleanupSessionTempBlocks(long sessionId, List<Long> tempBlockIds) {
    * @return available bytes
    * @throws IllegalArgumentException when location does not belong to tiered storage
    */
-  public long getAvailableBytes(BlockStoreLocation location) {
+  public long getAvailableBytes(BlockStoreLocation location) throws IllegalArgumentException {
```
Comment (apc999) [inline servers/src/main/java/tachyon/worker/block/BlockMetadataManager.java]: my view is `IllegalArgumentException` is a subclass of `java.lang.RuntimeException`, which we don't usually put into method signature. Do you have a particular reason of doing this, just to match the javadoc which declares throwing this exception? (if it is for the latter case,  I suggest to update the javadoc instead)

**Comment by apc999** [comment]
@HorizonNet thanks for the PR. I put a comment inline regarding `IllegalArgumentException`. Otherwise the patch looks good to me. Thanks!

**Comment by HorizonNet** [comment]
@apc999 Thanks for the comment. It was very helpful. I updated the PR to include the description for the ```IllegalArgumentException```  in the general documentation for the methods and removed the exception from the signature.

**Comment by HorizonNet** [comment]
Jenkins, test this please.

**Comment by apc999** [comment]
+1 LGTM

**Pull request merged**